pkg/sdk/connection.go

```diff
@@ -31,4 +31,8 @@ type Connection interface {
 	// Endpoint returns the endpoint (aka. base URL) of the Radius API. This definitely includes
 	// the URL scheme and authority, and may include path segments.
 	Endpoint() string
+
+	// InsecureAllowCredentialWithHTTP returns true if the connection allows sending credentials
+	// over HTTP. This is disabled by default.
+	InsecureAllowCredentialWithHTTP() bool
```

Comment:
Can you explain why this is needed? I don't want to clutter up the API with things we're doing.

Reply:
We have these tests: https://github.com/radius-project/radius/blob/main/pkg/portableresources/processors/resourceclient_test.go#L295.

And when we update the **azcore** package, these tests start to fail with this error: https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/runtime/policy_bearer_token.go#L120.

Here is where we create the necessary client for the tests mentioned above: https://github.com/radius-project/radius/blob/main/pkg/portableresources/processors/resourceclient.go#L182. I couldn't find any other way to set this flag to true just for the tests so that tests can pass.

Happy to discuss offline.

Reply:
Per discussion offline, can see if it's possible to remove the `BearerTokenPolicy`? None of the UCP scenarios should be using a bearer token, so we don't need this.